packages/runtime-core/src/helpers/resolveAssets.ts

```diff
@@ -33,6 +33,9 @@ export const NULL_DYNAMIC_COMPONENT = Symbol()
  */
 export function resolveDynamicComponent(component: unknown): VNodeTypes {
   if (isString(component)) {
+    // disambiguate between native element and component
+    if (isHTMLTag(component) || isSVGTag(component)) return component
```

Comment:
I think you should add `__COMPAT__` check inside this condition so it can be removed if compat mode is not enabled

```js
if (__COMPAT__ && (isHTMLTag(component) || isSVGTag(component)) return component
```

Also test should probably be moved to packages/vue-compat

Reply:
Although this issue sounds like it is a compat problem, in general case, dynamic component with native element as its tag name should always resolve to native element instead of a component. For some UI lib may register its component with  capitalized native element name, that may prevent dynamic component to resolve to be a native element in any case.